Modules/Microsoft365DSC/DSCResources/MSFT_EXOMailboxStatistics/MSFT_ExoMailboxStatistics.psm1

```diff
@@ -0,0 +1,1479 @@
+        $LegacyDN,
+
+        [Parameter()]
+        [System.String]
+        $MailboxMessagesPerFolderCountReceiveQuota,
+
+        [Parameter()]
+        [System.String]
+        $MailboxMessagesPerFolderCountWarningQuota,
+
+        [Parameter()]
+        [System.String]
+        $MailboxType,
+
+        [Parameter()]
+        [System.String]
+        $MailboxTypeDetail,
+
+        [Parameter()]
+        [System.String]
+        $MessageTableAvailableSize,
+
+        [Parameter()]
+        [System.String]
+        $MessageTableTotalSize,
+
+        [Parameter()]
+        [System.String]
+        $NamedPropertiesCountQuota,
+
+        [Parameter()]
+        [System.String]
+        $NeedsToMove,
+
+        [Parameter()]
+        [System.String]
+        $OtherTablesAvailableSize,
+
+        [Parameter()]
+        [System.String]
+        $OtherTablesTotalSize,
+
+        [Parameter()]
+        [System.String]
+        $OwnerADGuid,
+
+        [Parameter()]
+        [System.String]
+        $QuarantineClients,
+
+        [Parameter()]
+        [System.String]
+        $QuarantineDescription,
+
+        [Parameter()]
+        [System.String]
+        $QuarantineEnd,
+
+        [Parameter()]
+        [System.String]
+        $QuarantineFileVersion,
+
+        [Parameter()]
+        [System.String]
+        $QuarantineLastCrash,
+
+        [Parameter()]
+        [System.String]
+        $ResourceUsageRollingAvgDatabaseReads,
+
+        [Parameter()]
+        [System.String]
+        $ResourceUsageRollingAvgRop,
+
+        [Parameter()]
+        [System.String]
+        $ResourceUsageRollingClientTypes,
+
+        [Parameter()]
+        [System.String]
+        $ServerName,
+
+        [Parameter()]
+        [System.String]
+        $StorageLimitStatus,
+
+        [Parameter()]
+        [System.String]
+        $SystemMessageCount,
+
+        [Parameter()]
+        [System.String]
+        $SystemMessageSize,
+
+        [Parameter()]
+        [System.String]
+        $SystemMessageSizeShutoffQuota,
+
+        [Parameter()]
+        [System.String]
+        $SystemMessageSizeWarningQuota,
+
+        ########################
+        #Common parameters  
+        ########################
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [System.String]
+        $CertificatePath,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $CertificatePassword,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity
+    )
+
+    throw 'Updating Mailbox Statistics is not supported.'
```

Comment:
@ykuijs, I have inherited this PR. The customer who initiated the PR uses the mailbox statistic data to control the flow of a restore. Right, a resource that only reads data is not the primary goal of but there are examples that follow a similar pattern like [PendingReboot](https://github.com/dsccommunity/ComputerManagementDsc/tree/main/source/DSCResources/DSC_PendingReboot).

If you are supporting the idea in general I will make the requested code changes.

Reply:
PendingReboot actually changes something, it updates the $global:DSCMachineStatus variable.

I am leaning towards a feeling that M365DSC isn't the correct place for just retrieving information, especially for information that can be a very large data set (for environments that have large amounts of mailboxes, which can be the case quickly).

@NikCharlebois, what is your opinion on this resource?
(Responses might be a little delayed, since Nik is off at the moment)

Reply:
I would not implement this kind of change even though it is tempting to have the additional information in the exported data. #4375 describes a similar ask for a read-only property.